sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -1,107 +1,116 @@
+  AllowAuthorizedParty,
   HasLocationAccess,
   IsInstitutionAdmin,
-  AllowAuthorizedParty,
+  UserToken,
 } from "../../auth/decorators";
-import { AuthorizedParties } from "../../auth/authorized-parties.enum";
-import { OptionItem } from "../../types";
+import { IInstitutionUserToken } from "../../auth/userToken.interface";
 import {
-  ActiveApplicationSummaryDTO,
-  ActiveApplicationDataDto,
-} from "../application/models/application.model";
+  ApplicationService,
+  FormService,
+  InstitutionLocationService,
+} from "../../services";
+import { ClientTypeBaseRoute } from "../../types";
 import {
-  getUserFullName,
   dateString,
   getISODateOnlyString,
+  getUserFullName,
 } from "../../utilities";
-import { InstitutionLocation, Application } from "../../database/entities";
-import { ApiOkResponse, ApiTags } from "@nestjs/swagger";
-import { InstitutionLocationsControllerService } from "./institution-locations.controller.service";
+import {
+  ActiveApplicationDataAPIOutDTO,
+  ActiveApplicationSummaryAPIOutDTO,
+} from "../application/models/application.dto";
+import BaseController from "../BaseController";
+import { PrimaryIdentifierDTO } from "../models/primary.identifier.dto";
+import { InstitutionLocationControllerService } from "./institution-location.controller.service";
+import {
+  InstitutionLocationAPIOutDTO,
+  InstitutionLocationFormAPIInDTO,
+  InstitutionLocationFormAPIOutDTO,
+} from "./models/institution-location.dto";
 
+/**
+ * Institution location controller for institutions Client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
 @Controller("institution/location")
-@ApiTags("institution")
-export class InstitutionLocationsController extends BaseController {
+@ApiTags(`${ClientTypeBaseRoute.Institution}-institution`)
+export class InstitutionLocationInstitutionsController extends BaseController {
   constructor(
-    private readonly locationControllerService: InstitutionLocationsControllerService,
+    private readonly locationControllerService: InstitutionLocationControllerService,
     private readonly applicationService: ApplicationService,
     private readonly locationService: InstitutionLocationService,
     private readonly formService: FormService,
-    private readonly institutionService: InstitutionService,
   ) {
     super();
   }
 
-  @AllowAuthorizedParty(AuthorizedParties.institution)
+  /**
+   * Create an Institution location.
+   * @param payload
+   * @returns Primary identifier of created location.
+   */
+  @ApiBadRequestResponse({
+    description: "Invalid request to create an institution location.",
+  })
   @IsInstitutionAdmin()
   @Post()
   async create(
-    @Body() payload: InstitutionLocationTypeDto,
-    @UserToken() userToken: IUserToken,
-  ): Promise<number> {
+    @Body() payload: InstitutionLocationFormAPIInDTO,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<PrimaryIdentifierDTO> {
     // Validate the location data that will be saved to SIMS DB.
```

Comment:
As per the naming convention for DTOs this one needs to be renamed to PrimaryIdentifierAPIOutDTO.